src/index/TextIndexReadWrite.cpp

```diff
@@ -121,11 +121,12 @@ FrequencyEncode<T>::FrequencyEncode(const TypedVector& vectorToEncode) {
   // Convert the hashmap to a vector to sort it by most frequent first
   std::vector<std::pair<T, size_t>> frequencyVector;
   frequencyVector.reserve(frequencyMap.size());
-  std::transform(frequencyMap.begin(), frequencyMap.end(),
-                 std::back_inserter(frequencyVector),
-                 [](const auto& kv) { return kv; });
-  std::sort(frequencyVector.begin(), frequencyVector.end(),
-            [](const auto& a, const auto& b) { return a.second > b.second; });
+  ql::ranges::transform(frequencyMap.begin(), frequencyMap.end(),
+                        std::back_inserter(frequencyVector),
+                        [](const auto& kv) { return kv; });
+  ql::ranges::sort(
+      frequencyVector.begin(), frequencyVector.end(),
+      [](const auto& a, const auto& b) { return a.second > b.second; });
```

Comment:
```suggestion
  ql::ranges::transform(frequencyMap,
                        std::back_inserter(frequencyVector),
                        [](const auto& kv) { return kv; });
  ql::ranges::sort(
      frequencyVector,
      [](const auto& a, const auto& b) { return a.second > b.second; });
```
That's one of the points of `ql::ranges`.

And can't the first transorm with the identity function just simpler be

`ql::ranges::copy(frequeycMap, back_inserter(vector));`

Reply:
See b93312aad5a0366f93921d3c5610d002c3c24245